### PR TITLE
Bug 1812052: Update HACKING.md rebase to use go modules

### PR DESCRIPTION
**Pull request by juanluisvaladas**
I forgot to update HACKING.md after the migration to go modules.
This includes it.

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -71,62 +75,50 @@ make things easier...
...
+   First we need to update kubernetes itself, in go.mod, the dependencies on
+   the require section **must** point to a version that exists on that module
+   upstream. This means picking whatever is the closest to whatever you need,
+   for the k8s.io/* dependencies, for the 1.17.2 I used v1.17.2, but don't
+   worry too much about this because we'll actually define the dependency on
+   the replace section and `go mod tidy` will change whatever you write anyway.
+
+   The replace section is where we define what is the dependency which will
+   really be vendored. All the modules that are part of kubernetes must be
+   replaced to point to ther path in kubernetes. These depdendencies can be
```
Comment (JacobTanenbaum) [inline HACKING.md]: nit - "replaced to point to their"

File: HACKING.md [hunk 2]
```diff
@@ -144,14 +136,15 @@ version of `golang.org/x/sys` that was required by other
 dependencies.)
 
 Another thing that can happen (especially in the final rebase) is that
-if glide decides to pull in any new dependencies that aren't listed in
-`glide.yaml`, it may pull the git master version of it and get some
-code that depends on the master versions of everything else and so
-won't compile. In that case, the fix is to explicitly pin that new
-module to the right version in `glide.yaml`.
+if go mod decides to pull in any new dependencies that aren't listed in
+`go.mod` nor `go.lock`, it may pull a version incompatible with other
```
Comment (JacobTanenbaum) [inline HACKING.md]: nit - that aren't listed in 'go.mod' or 'go.lock'

File: HACKING.md [hunk 3]
```diff
@@ -10,38 +10,42 @@ make things easier...
    `sdn-${OPENSHIFT_RELEASE_NUMBER}-kubernetes-${KUBERNETES_RELEASE_NUMBER}`,
    based on the latest upstream release/pre-release tag. eg, the first
    rebase attempt for OCP 4.3 was the branch
-   `sdn-4.3-kubernetes-1.16.0-beta.2`, based off the upstream tag
-   `v1.16.0-beta.2`. The `${OPENSHIFT_RELEASE_NUMBER}` should just be
+   `sdn-4.4-kubernetes-1.17.2`, based off the upstream tag
+   `v1.17.2`. The `${OPENSHIFT_RELEASE_NUMBER}` should just be
```
Comment (danwinship) [inline HACKING.md]: The first half of the sentence is still talking about OCP 4.3

File: HACKING.md [hunk 4]
```diff
@@ -10,38 +10,42 @@ make things easier...
...
+            [...]
+            k8s.io/kubernetes => github.com/openshift/kubernetes v1.17.0-alpha.0.0.20190924141618-7eb200efda20
+            [...]
+         )
        ...
 
+   Unfortunately, go mod will autogenerate the version in a non human-friendly
+   way, in this case, `7eb200efda20` is the commit number. One way to see the
+   branch is going to the github.com/openshift/kubernetes repository, checkout
+   `7eb200efda20` and the use git log --oneline --decorate
```
Comment (danwinship) [inline HACKING.md]: you don't need to check it out, you can just `git log 7eb200efda20`

put backticks around the git command. And probably "go mod" above too.

File: HACKING.md [hunk 5]
```diff
@@ -10,38 +10,42 @@ make things easier...
...
+
    So, eg, [that branch in
-   `openshift/kubernetes`](https://github.com/openshift/kubernetes/commits/sdn-4.2-kubernetes-1.14.0)
+   [`openshift/kubernetes`](https://github.com/openshift/kubernetes/commits/sdn-4.3-kubernetes-1.16.0)
```
Comment (danwinship) [inline HACKING.md]: There's already an open bracket on the previous line. Don't add a second one here.

File: HACKING.md [hunk 6]
```diff
@@ -71,62 +75,50 @@ make things easier...
...
+       sed -i 's/v0.0.0-20190924141618-7eb200efda20/sdn-4.4-kubernetes-1.17.2/g' go.mod
 
-   That means that some part of `openshift/sdn` is trying to pull in
-   some package that no longer exists in one of its dependencies.
-   You'll need to figure out what happened with that package, and fix
-   our code accordingly. (eg, in this case,
-   `k8s.io/kubernetes/pkg/kubelet/apis/cri` moved to
-   `k8s.io/cri-api/pkg/apis`, so several files in `pkg/network/node`
-   had to be updated to point to the new place.)
+   You shouldn't need to update the rest of the dependencies unlesss there are
```
Comment (danwinship) [inline HACKING.md]: typo "unlesss"

File: HACKING.md [hunk 7]
```diff
@@ -71,62 +75,50 @@ make things easier...
...
-   the previous beta) or else see if `master` can reasonably be used.
-   If you do this leave a `# FIXME` comment in `glide.yaml`.
-
-   If you get to the final rebase (ie, `.0`, no `beta` or `rc`) and
-   there's some module where there's still no appropriate tag, then
-   it's better to depend on the commit SHA of the current master
-   rather than actually depending on `master`.
-
-2. Update the remaining kubernetes and openshift dependencies in
-   `glide.yaml`. For dependencies under `github.com/openshift/`, and
```
Comment (danwinship) [inline HACKING.md]: This step (non-kubernetes dependencies) seems to have been dropped?

**Comment by juanluisvaladas** [comment]
@danwinship can you please review again? the only changes are regarding your comments

**Pull request merged**